### PR TITLE
Register TPU regardless of success state of spam flag

### PR DESCRIPTION
**Pull request by iBug**
When someone clicks "Spam flag", it's sure they want to give a TPU feedback as well. The current code does not if the spam flag is not cast successfully. Moving relevant code to before flag attempt solves this.

### CODE REVIEW

**Comment by iBug** [comment]
Appears like that `rubocop -aD` modified Rakefile. Revert if it's bad (idk because I'm not familiar).

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -185,6 +185,12 @@ def delete_post
   end
 
   def cast_spam_flag
+    feedback = Feedback.new(feedback_type: 'tpu-',
+                            user_id: current_user.id,
+                            post_id: @post.id)
+
+    feedback.save || flash[:danger] = 'Unable to save feedback. Ping Undo.'
```
Comment (ArtOfCode-) [inline app/controllers/posts_controller.rb]: Prefer the original unless here, assign in a conditional feels icky

File: Rakefile [hunk 2]
```diff
@@ -3,6 +3,6 @@
 # Add your own tasks in files placed in lib/tasks ending in .rake,
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
-require File.expand_path('../config/application', __FILE__)
+require File.expand_path('config/application', __dir__)
```
Comment (ArtOfCode-) [inline Rakefile]: I don't think this needs changing, we can tell rubocop to shush here. Revert and ignore?

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (iBug) [reply app/controllers/posts_controller.rb]: @ArtOfCode- Yes I prefer the original unless too but rubocop doesn't think so.

Comment (iBug) [reply app/controllers/posts_controller.rb]: @ArtOfCode- Yes I prefer the original unless too but rubocop doesn't think so.

Comment (thesecretmaster) [reply app/controllers/posts_controller.rb]: I think rubocop would be equally happy with `flash... unless feedback.save`.

Comment (iBug) [reply app/controllers/posts_controller.rb]: @thesecretmaster I was thinking that it was too unintuitive (I did think of it, though) and was hesitant to write code in that way. Anyway I've done this as of now.

**Comment by thesecretmaster** [comment]
Looks good to me. Please hold on merging until this knot in yarn gets worked out though :)

**Comment by ArtOfCode-** [comment]
@thesecretmaster Is that knot cleared, or not yet?

**Pull request merged**